x/ibc/03-connection/client/utils/utils.go

```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+)
+
+// QueryConnection queries the store to get a connection end and a merkle
+// proof.
+func QueryConnection(
+	cliCtx context.CLIContext, connectionID string, prove bool,
+) (types.ConnectionResponse, error) {
+	req := abci.RequestQuery{
+		Path:  "store/ibc/key",
+		Data:  types.KeyConnection(connectionID),
+		Prove: prove,
+	}
+
+	res, err := cliCtx.QueryABCI(req)
+	if err != nil {
+		return types.ConnectionResponse{}, err
+	}
+
+	var connection types.ConnectionEnd
+	if err := cliCtx.Codec.UnmarshalBinaryLengthPrefixed(res.Value, &connection); err != nil {
+		return types.ConnectionResponse{}, err
+	}
+
+	connRes := types.NewConnectionResponse(connectionID, connection, res.Proof, res.Height)
+
+	return connRes, nil
+}
+
+// QueryClientConnections queries the store to get the registered connection paths
+// registered for a particular client and a merkle proof.
+func QueryClientConnections(
```

Comment:
Why do we need to prove this?

Reply:
it's not necessary, but @jackzampolin and I agreed that it wouldn't hurt to provide proofs for all queries in IBC. Do you have any thoughts?

Reply:
I want to have full coverage for any proofs that will be used in a transaction.

Reply:
I want to have full coverage for any proofs that will be used in a transaction.

Reply:
Is this used in a transaction? Regardless, having a proof is fine.

Reply:
> Is this used in a transaction? Regardless, having a proof is fine.

not really, we saw it as a nice-to-have